.github/workflows/build.yml

```diff
@@ -77,25 +82,35 @@ jobs:
                   build_doc = "false"
           oses = ["macos", "ubuntu", "windows"]
           build_dict = {os : [k for k in build if k.startswith(os)] for os in oses}
-          print(f"::set-output name=build::{build_dict}")
-          print(f"::set-output name=test::{dict({os : [k for k in test if k.startswith(os)] for os in oses})}")
-          print(f"::set-output name=build_doc::{build_doc}")
-          for os in oses:
-              print(f"::set-output name=do_{os}::{'true' if len(build_dict[os]) > 0 else 'false'}")
-          print(f"::set-output name=python_version::{python_version}")
+
+          with open(environ["GITHUB_OUTPUT"], "a") as f:
+              f.write(f"build={dumps(build_dict)}\n")
+              f.write(f"test={dumps(dict({os : [k for k in test if k.startswith(os)] for os in oses}))}\n")
+              f.write(f"build_doc={dumps(build_doc)}\n")
+              for os in oses:
+                  f.write(f"do_{os}={dumps('true' if len(build_dict[os]) > 0 else 'false')}\n")
+              f.write(f"python_version={dumps(python_version)}\n")
 
   lint-sources:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
         with:
           python-version: "3.8"
-      - uses: pre-commit/action@v2.0.0
+      - name: install pre-commit
+        run: python -m pip install pre-commit
+      - name: get cached pre-commit hooks
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}
+      - name: pre-commit checks
+        run: pre-commit run --show-diff-on-failure --color=always
 
   build-windows:
     needs: [setup]
-    if: needs.setup.outputs.do_windows == 'true'
+    if: needs.setup.outputs.do_windows
```

Comment:
Are you sure that this change does not break anything?  AFAICT you did not test it.

Reply:
Moved back to the previous statement.